manager/scheduler/constraint.go

```diff
@@ -44,26 +44,36 @@ func (f *ConstraintFilter) Check(n *NodeInfo) bool {
 			if !constraint.Match(n.Description.Hostname) {
 				return false
 			}
+		case "node.role":
+			if !constraint.Match(n.Spec.Role.String()) {
+				return false
+			}
 		default:
 			// default is node label in form like 'node.labels.key==value'
 			// if it is not well formed, always fails it
 			if !strings.HasPrefix(constraint.Key, "node.labels.") {
 				return false
 			}
-			// if the node doesn't have any label,
-			// it's equivalent to match an empty value.
-			// that is, 'node.labels.key!=value' should pass and
-			// 'node.labels.key==value' should fail
-			if n.Spec.Annotations.Labels == nil {
-				if !constraint.Match("") {
-					return false
+
+			// node has labels from node.Spec.Annotations.Labels and node.Description.Engine.Labels
+			// if a label exists on both, node.Spec.Annotations.Labels has precedence
+			combinedLabels := make(map[string]string)
+			if n.Description != nil && n.Description.Engine != nil && n.Description.Engine.Labels != nil {
```

Comment:
@diogomonica @stevvooe Thanks for bringing up this use case. 

We have Engine labels and NodeSpec labels. Engine labels typically contain resource related info. For example, some nodes have GPU, SSD. VM bootstrap can automatically labels the engines. NodeSpec are controlled by system administrators. 

Current code doesn't differentiate them. One simple fix is to match Engine labels with `--constraint engine.labels.key==value`, and NodeSpec labels with `--constraint node.labels.key==value`.

cc @aluzzardi .